### ISSUE TITLE
Restarting homu when the tree is closed bricks the web UI

### ISSUE THREAD

**Issue by pietroalbini** [rust-lang/homu#50]
We had to restart the production homu instance while the tree was closed, and once we restarted it the web UI raised this exception:

```
Traceback (most recent call last):                                                                                              
  File "/usr/local/bin/bottle.py", line 862, in _handle                                                                         
    return route.call(**args)                                                                                                   
  File "/usr/local/bin/bottle.py", line 1742, in wrapper                                                                        
    rv = callback(*a, **ka)                                                                                                     
  File "/homu/homu/server.py", line 75, in index                                                                                
    for label in sorted(g.repos)])                                                                                              
  File "/usr/local/lib/python3.6/dist-packages/jinja2/asyncsupport.py", line 76, in render                                      
    return original_render(self, *args, **kwargs)                                                                               
  File "/usr/local/lib/python3.6/dist-packages/jinja2/environment.py", line 1008, in render                                     
    return self.environment.handle_exception(exc_info, True)                                                                    
  File "/usr/local/lib/python3.6/dist-packages/jinja2/environment.py", line 780, in handle_exception                            
    reraise(exc_type, exc_value, tb)                                                                                            
  File "/usr/local/lib/python3.6/dist-packages/jinja2/_compat.py", line 37, in reraise                                          
    raise value.with_traceback(tb)                                                                                              
  File "/homu/homu/html/index.html", line 26, in top-level template code                                                        
    <li><a href="queue/{{repo.repo_label}}">{{repo.repo_label}}</a> {% if repo.treeclosed >= 0 %} [TREE CLOSED] {% endif %}</li>
jinja2.exceptions.UndefinedError: 'github3.repos.repo.Repository object' has no attribute 'treeclosed'                          
```

To "fix" the issue I had to reopen the tree and then restart homu again. We should properly fix this.

**Comment by bryanburgers** [comment]
I was unable to reproduce this locally. I tried a few different ways, and couldn't get it to happen.

However, the exception says that our repo object when this failed is a 'github3.repos.repo.Repository' object (one from the github3 library) instead of what we'd expect: a 'Repository' object as defined in main.py.

This code looks suspicious: https://github.com/rust-lang/homu/blob/master/homu/server.py#L62-L69

```
def get_repo(repo_label, repo_cfg):
    repo = g.repos[repo_label].gh
    if not repo:
        repo = g.gh.repository(repo_cfg['owner'], repo_cfg['name'])
        g.repos[repo_label] = repo
```

That last line looks like we're assigning to `g.repos[repo_label]` a `github3.*.Repository` instead of our own `Repository`.

Compare with a very similar snippet from main.py: https://github.com/rust-lang/homu/blob/master/homu/main.py#L306-L314

```
def get_repo(self):
    repo = self.repos[self.repo_label].gh
    if not repo:
        repo = self.gh.repository(self.owner, self.name)
        self.repos[self.repo_label].gh = repo
```

(I wish somebody would make a language that could catch these sorts of type errors...)

My theory is that _somehow_ this is a race condition where server.py's `get_repo` gets called before main.py's `get_repo`. But I'm not able to confirm that.

**Comment by pietroalbini** [comment]
This *should* be fixed by #51.
